components/profile-controller/controllers/profile_controller.go

```diff
@@ -114,8 +116,11 @@ func (r *ProfileReconciler) Reconcile(request ctrl.Request) (ctrl.Result, error)
 		ObjectMeta: metav1.ObjectMeta{
 			Annotations: map[string]string{"owner": instance.Spec.Owner.Name},
 			// inject istio sidecar to all pods in target namespace by default.
-			Labels: map[string]string{"istio-injection": "enabled"},
-			Name:   instance.Name,
+			Labels: map[string]string{
+				istioInjectionLabel:        "enabled",
+				katibMetricsCollectorLabel: "enabled",
```

Comment:
Seems the label value is always "enabled".
Then shall we just inject katib metrics collector by default to all katib pods?

Reply:
metrics collector is injected into katib pods as a sidecar container by mutating webhook, for performance consideration, only the pods whose namespace has "katib-metricscollector-injection" label will be handled.

Reply:
Can we instead have a 
`kubeflow-managed`: `true`
label on all kubeflow namespaces?

I feel other controllers will need similar label when kubeflow got deployed as part of a giant cluster.

Reply:
I am not sure
katib can work well all alone, even not as a component of kubeflow

Reply:
@jlewi what's your preference?

Is `katib namespace` == `kubeflow namespace`?
If they are same set, then we can use 1 label for multiple controllers.

Reply:
Will istio inject the istio sidecar to every pod in the namespace if we set it to enabled? If that, it may affect the behavior of katib. When users use job to run HPO jobs, we assume that containers[0] in the job is the actual training job.

Reply:
@kunmingg This label will be removed after k8s 1.15 becomes the default. See related comment https://github.com/kubeflow/kubeflow/issues/4231#issuecomment-539305378

Reply:
I think we need to establish guidelines around how admission hooks should behave.

I can see reasons for both arguments

* Application specific labels make it easy to selectively enable/disable an application web hook on a namespace level

On the other hand we could potentially get an explosion of labels if every application adds its own.

I'd probably lean towards being able to selectively disable/enable katib on a per namespace level. That feels more defensive. 

Katib is still Alpha so I think we can change the behavior in subsequent releases. So I don't have a strong oppinion either way.

Reply:
> I'd probably lean towards being able to selectively disable/enable katib on a per namespace level. That feels more defensive.

I agree this allows for a cluster-admin to switch preferences for each namespace.

Reply:
@johnugeorge 
1.15 is required for multi version CR as well, looking forward to it